### PR TITLE
mcux: MCXW71 plumbing

### PR DESCRIPTION
**Pull request by decsny**
Add code to build MCXW71 in zephyr.

### CODE REVIEW

File: mcux/README [hunk 1]
```diff
@@ -91,9 +91,12 @@ Patch List:
   - Patched mcux-sdk\components\conn_fwloader\mcux-sdk\components\conn_fwloader\driver_conn_fwloader.cmake for TFM support
   - mcux-sdk/drivers/s3mu/fsl_s3mu.c: rename _BIT macro to avoid collision from util_macro.h in Zephyr
   - Modifed mcux\mcux-sdk\drivers\mipi_dsi_split\fsl_mipi_dsi.c line 541 to force 8.0 to float (8.0f) to avoid [-Double-promotion] warning
-  - Wrap conditional code setting SCB->VTOR with !defined(__ZEPHYR__) in system c files of the folloing parts: 
+  - Wrap conditional code setting SCB->VTOR with !defined(__ZEPHYR__) in system c files of the folloing parts:
```
Comment (dleach02) [inline mcux/README:94]: I guess we don't have check-compliance on the HAL... looks like you trimmed spaces. Can you also fix the "folloing" to be "following"?

Comment (decsny) [reply mcux/README:94]: interesting, i didnt even realize i did that, i put an autocommand on my editor to remove the spaces at ends of lines when i save the file

**Comment by danieldegrasse** [comment]
@mmahadevan108 do we need to add this as a patch to the README? Not sure what the procedure is for new devices

**Comment by mmahadevan108** [comment]
> @mmahadevan108 do we need to add this as a patch to the README? Not sure what the procedure is for new devices
If the devices is already added to the SDK repo then I would not worry about adding to the README. Updating this is more important for patches that are maintained in our HAL and not integrated into the SDK repo.

**Review by danieldegrasse — APPROVED** [review]

File: dts/nxp/mcx/MCXW716CMFPA-pinctrl.h [hunk 2]
```diff
@@ -0,0 +1,180 @@
+/*
+ * NOTE: Autogenerated file by gen_soc_headers.py
+ * for MCXW716CMFPA/signal_configuration.xml
+ *
```
Comment (danieldegrasse) [inline dts/nxp/mcx/MCXW716CMFPA-pinctrl.h:4]: File needs copyright

File: dts/nxp/mcx/MCXW716CMFTA-pinctrl.h [hunk 3]
```diff
@@ -0,0 +1,228 @@
+/*
+ * NOTE: Autogenerated file by gen_soc_headers.py
+ * for MCXW716CMFTA/signal_configuration.xml
+ *
+ *
```
Comment (danieldegrasse) [inline dts/nxp/mcx/MCXW716CMFTA-pinctrl.h:5]: Ditto, generate these files with the `--copyright` switch

**Comment by decsny** [comment]
> > @mmahadevan108 do we need to add this as a patch to the README? Not sure what the procedure is for new devices
> > If the devices is already added to the SDK repo then I would not worry about adding to the README. Updating this is more important for patches that are maintained in our HAL and not integrated into the SDK repo.

to be clear, the device is in the sdk but they forgot these files

File: dts/nxp/mcx/MCXW716CMFPA-pinctrl.h [hunk 4]
```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2024 NXP
```
Comment (danieldegrasse) [inline dts/nxp/mcx/MCXW716CMFPA-pinctrl.h:2]: Missing the SPDX License header. Did you generate this file, or add the copyright manually? I think the generation script will make this file Apache-2.0 licensed, so let's use that

**Review by danieldegrasse — CHANGES_REQUESTED** [review]
License on pinctrl files still needs SPDX tag

File: dts/nxp/mcx/MCXW716CMFPA-pinctrl.h (hunk 4 above)
Comment (decsny) [reply dts/nxp/mcx/MCXW716CMFPA-pinctrl.h:2]: updated

**Review by danieldegrasse — APPROVED** [review]

**Review by mmahadevan108 — CHANGES_REQUESTED** [review]
Will Block till the Zephyr review is close to complete.

**Comment by decsny** [comment]
> Will Block till the Zephyr review is close to complete.

the only thing holding up the zephyr PR is that every 4 hours i have to fix a conflict in the hal

**Comment by decsny** [comment]
update: fixed conflict

**Review by EmilioCBen — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**